spec/abci/abci.md

```diff
@@ -5,27 +5,32 @@ title: Method and Types
 
 # Methods and Types
 
-## Overview
-
-The ABCI message types are defined in a [protobuf
-file](https://github.com/tendermint/tendermint/blob/master/proto/tendermint/abci/types.proto).
-
-ABCI methods are split across four separate ABCI _connections_:
-
-- Consensus connection: `InitChain`, `BeginBlock`, `DeliverTx`, `EndBlock`, `Commit`
-- Mempool connection: `CheckTx`
-- Info connection: `Info`, `Query`
-- Snapshot connection: `ListSnapshots`, `LoadSnapshotChunk`, `OfferSnapshot`, `ApplySnapshotChunk`
-
-The consensus connection is driven by a consensus protocol and is responsible
-for block execution.
-
-The mempool connection is for validating new transactions, before they're
-shared or included in a block.
-
-The info connection is for initialization and for queries from the user.
-
-The snapshot connection is for serving and restoring [state sync snapshots](apps.md#state-sync).
+## Connections
+
+ABCI applications can run either within the same _process_ as the Tendermint 
+state-machine replication engine _or_ as a separate process from the state-machine
+replication engine. When run within the same process, Tendermint will call the ABCI
+application methods directly as typical Go method calls. When Tendermint and the
+ABCI application are run as separate processes, Tendermint opens a set of connections
+to the application.
+
+When Tendermint and the application are run as separate processes, Tendermint opens
+four connections for ABCI methods. The connections each handle a subset of the ABCI
+method calls. They are defined as follows:
```

Comment:
There is some duplication between these two paragraphs that I'd suggest collapsing.
```suggestion
application methods directly as Go method calls.

When Tendermint and the ABCI application are run as separate processes, Tendermint
opens four connections to the application for ABCI methods. The connections each 
handle a subset of the ABCI method calls. These subsets are defined as follows:
```

Reply:
Not directly related to this change, but it seems odd to have all of these on separate connections. Is there some rationale for that, which we could summarize? (Made-up hypothetical: "This allows the application to more easily handle different parts of the consensus workflow concurrently" or something like that?)

Reply:
I'm not 100% sure I know the answer. this document is how I learned about how this works. I'd recommend asking in the tendermint slack channel.